lib/srv/sess.go

```diff
@@ -725,6 +733,17 @@ func (m *multiWriter) deleteWriter(id string) {
 	delete(m.writers, id)
 }
 
+func (m *multiWriter) lockedAddRecentWrite(p []byte) {
```

Comment:
yeah, I see this pattern inside Golang's own runtime source code, I'd propose we adopt it internally too.